config/fields/structure.php

```diff
@@ -108,55 +108,44 @@
 			return $this->form()->fields()->toArray();
 		},
 		'columns' => function () {
-			$columns = [];
-			$mobile  = 0;
-
-			if (empty($this->columns) === true) {
-				foreach ($this->fields as $field) {
-					// Skip hidden and unsaveable fields
-					// They should never be included as column
-					if (
-						$field['type'] === 'hidden' ||
-						$field['hidden'] === true ||
-						$field['saveable'] === false
-					) {
-						continue;
-					}
+			$columns   = [];
+			$blueprint = $this->columns;
+
+			// if no custom columns have been defined,
+			// gather all fields as columns
+			if (empty($blueprint) === true) {
+				$fields    = array_column($this->fields, 'name');
+				$blueprint = array_fill_keys($fields, true);
+			}
 
-					$columns[$field['name']] = [
-						'type'  => $field['type'],
-						'label' => $field['label'] ?? $field['name']
-					];
+			foreach ($blueprint as $name => $column) {
+				$field = $this->fields[$name] ?? null;
+
+				// Skip empty, hidden and unsaveable fields
+				// They should never be included as column
+				if (
+					empty($field) === true ||
+					$field['type'] === 'hidden' ||
+					$field['hidden'] === true ||
```

Comment:
Could there be a use case where a hidden field was explicitly defined as a column? Previously this worked, now the column will be hidden.

Reply:
Good catch, but thinking about it I have a hard time to imagine a use case where the field inside the drawer is supposed to be hidden but then shows up in the preview table. 

But in general open to either (allowing it or mentioning as breaking change).

Reply:
Maybe it could be used for auto-generated information that should be visible in the table but does not need to be edited and is therefore hidden from the form.

If the code could be easily modified to allow this, I'd prefer that. Otherwise I'm also fine with mentioning it as a breaking change. If users are affected by this, they will surely tell us so.

Reply:
I'd think that even for that a disabled field would make more sense in the drawer than hiding it.

But I added a commit that supports it, I think. Just filters hidden fields from the default columns, but allows them when columns have been specified.